android/app/src/main/java/com/dismoi/scout/floating/layout/Message.kt

```diff
@@ -1,20 +1,157 @@
 package com.dismoi.scout.floating.layout
 
+import android.animation.*
+import android.app.Service
 import android.content.Context
+import android.graphics.BitmapFactory
+import android.os.Build
+import android.os.StrictMode
+import android.text.Html
+import android.text.SpannableStringBuilder
+import android.text.Spanned
+import android.text.method.LinkMovementMethod
+import android.text.style.ClickableSpan
+import android.text.style.URLSpan
 import android.util.AttributeSet
-import android.animation.*
+import android.util.Log
+import android.view.LayoutInflater
 import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.annotation.RequiresApi
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import com.dismoi.scout.R
+import com.synnapps.carouselview.CarouselView
+import com.synnapps.carouselview.ViewListener
+import org.json.JSONObject
+import java.net.URL
 
+@RequiresApi(Build.VERSION_CODES.N)
 class Message(context: Context, attrs: AttributeSet?) : Layout(context, attrs) {
+  // TODO have correct class for notices
+  var notices: List<MutableLiveData<JSONObject>> = listOf()
+    set (notices) {
+      val carouselView = findViewById<CarouselView>(R.id.carouselView)
+      carouselView.pageCount = notices.size
+      field = notices
+    }
+
+  private var inflater: LayoutInflater = context.getSystemService(Service.LAYOUT_INFLATER_SERVICE) as LayoutInflater
+
+  // TODO Fix fonts usage
+  //    val type = Typeface.createFromAsset(context.assets, "fonts/Helvetica.ttf")
+  //    val typeBold = Typeface.createFromAsset(context.assets, "fonts/Helvetica-Bold.ttf")
+
+  private var carouselListener = ViewListener { position ->
```

Comment:
This method is soooo long, this would deserve to be break down in multiple function to be better understood.

Reply:
Totally ! 

Reply:
#245